src/Build/Microsoft.Build.csproj

```diff
@@ -30,6 +30,7 @@
 
   <ItemGroup>
     <ProjectReference Include="..\Framework\Microsoft.Build.Framework.csproj" />
+    <ProjectReference Include="..\Tasks\Microsoft.Build.Tasks.csproj" />
```

Comment:
I wouldn't make this dependency. The engine depending on tasks sounds like it would enable accidental engine dependency on task state. If it can happen at some point it will probably happen.

Reply:
Can we break this via reflection? As a process-startup cost it doesn't seem too bad, and would have most of the beneficial effects Mihai mentions.

Reply:
Ok, removed reference to the Tasks project and changed it to reflection.